mne/viz/evoked.py

```diff
@@ -601,17 +601,17 @@ def _plot_image(data, ax, this_type, picks, cmap, unit, units, scalings, times,
     ax.set(ylabel=ylabel, xlabel='Time (%s)' % (time_unit,), title=t)
     _add_nave(ax, nave)
 
-    if show_names is not False:
-        if show_names == "all":
-            yticks = np.arange(len(picks)).astype(int)
-            yticklabels = np.array(ch_names)[picks]
-        else:
-            max_tick = len(picks)
-            yticks = [tick for tick in ax.get_yticks() if tick < max_tick]
-            yticks = np.array(yticks).astype(int)
-            # these should only ever be ints right?
+    if show_names == "all":
+        yticks = np.arange(len(picks)).astype(int)
+        yticklabels = np.array(ch_names)[picks]
+    else:
+        yticks = np.round(ax.get_yticks()).astype(int)
+        yticks = np.intersect1d(yticks, np.arange(len(picks), dtype=int))
+        if show_names:
             yticklabels = np.array(ch_names)[picks][yticks]
-        ax.set(yticks=yticks + .5, yticklabels=yticklabels)
+        else:
+            yticklabels = np.array(picks)[yticks]
```

Comment:
I would right now tend to factor out this logic into two functions `get_yticklabels` and `get_yticks`.

Reply:
I refactored it down to just 5 lines, so at that point it didn't seem necessary to make functions anymore.